src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs

```diff
@@ -25,14 +26,55 @@ public static void EnsureDotnet()
             }
         }
 
+        public static async Task<int> GetDotnetMajorVersion()
+        {
+            EnsureDotnet();
+            Version dotnetVersion;
+            var exe = new Executable("dotnet", "--version");
+            var stdout = new StringBuilder();
+            var exitCode = await exe.RunAsync(o => stdout.AppendLine(o), e => ColoredConsole.Error.WriteLine(ErrorColor(e)));
+            // We split based on the the '-' character and take the first element to ensure that version parsing still works with preview/RC verisons of dotnet
+            var dotnetVersionString = stdout.ToString().Split("-")[0];
+            if (exitCode != 0)
+            {
+                throw new CliException("Error running dotnet --version");
+            }
+            else if (!Version.TryParse(dotnetVersionString, out dotnetVersion))
+            {
+                throw new CliException("dotnet version unable to be parsed!");
+            }
+            return dotnetVersion.Major;
+        }
+
+        public static async Task<string> GetTargetFramework()
+        {
+            var framework = "net";
+            var dotnetMajorVersion = await GetDotnetMajorVersion();
+            if (dotnetMajorVersion == 7)
+            {
+                framework += "7.0";
+            }
+            else if (dotnetMajorVersion == 6)
+            {
+                framework += "6.0";
+            }
+            else
+            {
+                framework += "48";
+            }
+            return framework;
+        }
+
         public async static Task DeployDotnetProject(string Name, bool force, WorkerRuntime workerRuntime)
         {
             await TemplateOperation(async () =>
             {
+                var shortName = "func";
+                var framework = await GetTargetFramework();
                 var connectionString = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
                     ? $"--StorageConnectionStringValue \"{Constants.StorageEmulatorConnectionString}\""
                     : string.Empty;
-                var exe = new Executable("dotnet", $"new func --AzureFunctionsVersion v4 --name {Name} {connectionString} {(force ? "--force" : string.Empty)}");
+                var exe = new Executable("dotnet", $"new {shortName} --Framework {framework} --AzureFunctionsVersion v4 --name {Name} {connectionString} {(force ? "--force" : string.Empty)}");
```

Comment:
the short name will be func for everything still, so you probably don't need to change that part